src/foam/dao/AuthenticatedSink.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.dao;
+
+import foam.core.Detachable;
+import foam.core.FObject;
+import foam.nanos.auth.AuthService;
+
+import javax.security.auth.AuthPermission;
+import java.security.Permission;
+
+public class AuthenticatedSink
+    extends ProxySink
+{
+  protected String name_;
```

Comment:
These two properties could be combined into just 'prefix'. Not too concerned either way.

Reply:
Done